### DISCUSSION TITLE
auto-sync with existing Heroku database URLs

### DISCUSSION THREAD

**Discussion by takeshiemoto** [hasura/graphql-engine#7500] [Question] [answered]
Is there any way to auto-sync database URLs after Hasura Cloud is connected to an existing Heroku database?

When I created a database on Heroku via Hasura Cloud, autosync was set up by default, but I don't know how to autosync it with an existing database.

**Comment by BenoitRanque** [comment, marked as answer]
**Edit**: This is now possible from the cloud dashboard, under integrations
~~Opting-in to the database url sync isn't currently possible.~~
As a work-around, you can follow these steps:

1. use the console to add another database, this time using the "create heroku database" option
2. copy the existing database to the newly created application.

This can be done using the heroku cli, with the following command:
```
heroku pg:copy app_name_to_copy_from::database_color_to_copy_from database_color_to_copy_to --app app_name_to_copy_to
```
The `pg:copy` cmdlet is documented [here](https://devcenter.heroku.com/articles/heroku-cli-commands#heroku-pg-copy-source-target)

Edit: I can confirm the ability to opt-in to the database url sync is on the roadmap.

**Reply by takeshiemoto** [reply]
Thanks for the answer!

Thanks also for the workaround suggestions.

Where can I find the roadmap?

**Reply by BenoitRanque** [reply]
I'm afraid this isn't currently on the public roadmap.
A WIP PR exists, and is being actively worked on, but it is on a private repo so I cannot share it here.
Unless something unexpected happens, the feature should land if not this month, then the next.

**Comment by hiran-hiran** [comment]
@BenoitRanque 
We expect this feature.
How's the progress?

**Reply by martin-hasura** [reply]
Hey there - this feature has been released and is currently available under `Project -> Integrations` under Cloud Dashboard. 
If you link your Heroku account you will be presented with a list of databases to sync to an env variable (you will then be able to take that variable and use it as a data source in Console).

We'll be doing some work to expose it more effectively from the design side in Console as well.

**Comment by hiran-hiran** [comment]
@martin-hasura 
Thank you for your response!

But, the way you suggested have to create a new Hasura-project.
I have no synchronized Hasura-project and Heroku-DB, so I want to synchronize existing both app.

Is there any way to do this?

**Reply by BenoitRanque** [reply]
@hiran-hiran this is currently possible, you can do this by going to your project settings in the cloud dashboard (not the console), and from there the heroku integration. You do not need to create a new hasura project.

**Reply by hiran-hiran** [reply]
@BenoitRanque 
It didn't display on Integrations in the dashboard.
Is it need owner role?

**Reply by hiran-hiran** [reply]
Tahnks!
I changed owner role, it worked.
So, I tried to do the same actions for other App, then I've got another problem.
`This Heroku database is not provisioned with Heroku Postgres.`

How can I solve this problem?

**Reply by hiran-hiran** [reply]
I was able to connect both apps.
I had to add `DATABASE_URL` to Config Vars on Heroku dashboard.

Thank you for your kind support.
